packages/web-components/fast-foundation/src/tree-item/tree-item.ts

```diff
@@ -192,11 +179,15 @@ export class FASTTreeItem extends FASTElement {
     public childItemLength(): number {
         const treeChildren: HTMLElement[] = this.childItems.filter(
             (item: HTMLElement) => {
-                return isTreeItemElement(item);
+                return FASTTreeItem.isFASTTreeItemElement(item);
             }
         );
         return treeChildren ? treeChildren.length : 0;
     }
+
+    public static isFASTTreeItemElement(el: Element): el is HTMLElement {
+        return isHTMLElement(el) && el instanceof FASTTreeItem;
```

Comment:
One trick about this is that `instanceof` will fail if the `el` has not yet been upgraded. So, this check succeeding assumes that the tree item is defined with the platform before this code runs. It seems like that's a safe bet since it's checking against itself, which means the upgrade must have already happened. However, I want to point this out because this pattern could fail if it was one element type checking another element type. So, we should be careful about applying this implementation if the other scenario comes up.

Reply:
Frankly I'd prefer to use the original method and figure out why the role isn't attached to the element when it's in the template. I'm not sold on this implementation as a fix yet, but I did want to try and resolve the issue for this specifically (rather than a broad approach...) because the current code executed _must_ match tagName which kind of kills the point of using this with a custom name.